### PR TITLE
Implement EinsplineSetT template class

### PR DESCRIPTION
**Pull request by PhilipFackler**
## Proposed changes

Implement EinsplineSetT

## What type(s) of changes does this code introduce?

- Refactoring (no functional changes, no api changes)

### Does this introduce a breaking change?

- No

## What systems has this change been tested on?
Ubuntu 22.04

## Checklist

- Yes. This PR is up to date with current the current state of 'develop'
- Yes. Code added or changed in the PR has been clang-formatted
- No. This PR adds tests to cover any new code, or to catch a bug that is being fixed
- No. Documentation has been added (if appropriate)

### CODE REVIEW

File: src/QMCWaveFunctions/EinsplineSetT.cpp [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+EinsplineSetT<T>::resetSourceParticleSet(ParticleSet& ions)
+{
+}
+
+template <typename T>
+void
+EinsplineSetT<T>::setOrbitalSetSize(int norbs)
+{
+	this->OrbitalSetSize = norbs;
+}
```
Comment (williamfgc) [inline src/QMCWaveFunctions/EinsplineSetT.cpp:48]: Are we missing concretizing classes? `template class EinsplineSetT<double>` etc.

File: src/QMCWaveFunctions/EinsplineSetT.h [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+template <typename T>
+class EinsplineSetT : public SPOSetT<T>
+{
+public:
+	//////////////////////
+	// Type definitions //
+	//////////////////////
+    using PosType = typename SPOSetT<T>::PosType;
+    using RealType = typename SPOSetT<T>::RealType;
+	using FullType = typename OrbitalSetTraits<double>::RealType;
```
Comment (williamfgc) [inline src/QMCWaveFunctions/EinsplineSetT.h]: clang-format

**Review by williamfgc — CHANGES_REQUESTED** [review]

File: src/QMCWaveFunctions/EinsplineSetT.cpp (hunk 1 above)
Comment (PhilipFackler) [reply src/QMCWaveFunctions/EinsplineSetT.cpp:48]: yes XD

File: src/QMCWaveFunctions/EinsplineSetT.h (hunk 2 above)
Comment (PhilipFackler) [reply src/QMCWaveFunctions/EinsplineSetT.h]: done

File: src/QMCWaveFunctions/EinsplineSetT.cpp [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+#include "CPU/e2iphi.h"
+#include "CPU/math.hpp"
+#include "einspline/multi_bspline.h"
+#include "type_traits/ConvertToReal.h"
+
+namespace qmcplusplus
+{
+template <typename T>
+EinsplineSetT<T>::EinsplineSetT(const std::string& my_name) :
+	SPOSet(my_name),
```
Comment (williamfgc) [inline src/QMCWaveFunctions/EinsplineSetT.cpp]: Should be SPOSetT<T>

Comment (williamfgc) [reply src/QMCWaveFunctions/EinsplineSetT.cpp]: Try compiling locally with SPOSetT<T>

**Review by williamfgc — CHANGES_REQUESTED** [review]

File: src/QMCWaveFunctions/EinsplineSetT.cpp (hunk 3 above)
Comment (PhilipFackler) [reply src/QMCWaveFunctions/EinsplineSetT.cpp]: done

**Comment by williamfgc** [comment]
Not needed anymore since EinsplineSet is being removed in [QMCPACK/qmcpack#4688](https://github.com/QMCPACK/qmcpack/pull/4688)

**Pull request closed without merging**